sig-scalability/goals.md

```diff
@@ -0,0 +1,78 @@
+# Kubernetes Scaling and Performance Goals
+
+_by Quinton Hoole and Wojciech Tyczynski, Google Inc_
+
+**April 2016**
+
+This document is a markdown version converted from a working [Google Doc](https://docs.google.com/document/d/1pABjn69yeDJDl0Yrsneiwj5A_bUI7BZgvKTQtSwzjiE/edit).  Please refer to the original for extended commentary and discussion.
+
+## Introduction
+
+What size clusters do we think that we should support with Kubernetes in the short to medium term?  How performant do we think that the control system should be at scale?  What resource overhead should the Kubernetes control system reasonably consume?  This document attempts to answer these and related questions.
+
+## Background
+Based on our experience running very large production systems at Google, we have found that running excessively large compute and/or storage clusters has several downsides, most notably:
+
+1. **Large-scale cluster failures can lead to large losses of capacity**: Despite our best efforts, correlated cluster-wide hardware or software failures do happen occasionally.  Even with built-in redundancy, load shedding, and good system design practises, the reality is that data center power supplies, and their backup systems do sometimes fail at the same time. Software rollouts, reconfigurations, system overloads and other cascading failures do occasionally go wrong at cluster-wide scale.  When they do (not if they do), it's better not to lose all (or even half of) your total capacity.
```

Comment:
I think we should augment some of the comments.  We've already received inquires from folks who would like to push kube to enable some HPC-esk type of workload support.  Typically these environments care less about web-uptime vs. compute utilization of very large single cluster deployments.  

/cc @jeremyeder @ConnorDoyle